### PR TITLE
Migrate to redux-persist v5.

### PR DESCRIPTION
**Pull request by Amurmurmur**
Bump version to 0.1.0.
Breaking changes.

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -72,6 +59,30 @@ const add = async function (context) {
...
+    insert: `import { persistReducer } from 'redux-persist'`,
+    after: `from 'redux'`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Redux/index.js`, {
+    insert: `import ReduxPersist from '../Config/ReduxPersist'`,
+    after: `from '../Sagas/'`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Redux/index.js`, {
+    insert: `
+    let finalReducers = reducers
```
Comment (ryanlntn) [inline plugin.js]: It looks like there are some whitespace issues here. These need to be unindented by two spaces to be formatted properly. Right now it looks like this:

image

A newline at the end would be nice too. 😄

File: plugin.js [hunk 2]
```diff
@@ -72,6 +59,30 @@ const add = async function (context) {
...
+    // If rehydration is on use persistReducer otherwise default combineReducers
+    if (ReduxPersist.active) {
+      const persistConfig = ReduxPersist.storeConfig
+      finalReducers = persistReducer(persistConfig, reducers)
+    }`,
+    after: `export default`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Containers/RootContainer.js`, {
```
Comment (ryanlntn) [inline plugin.js]: It looks like this should be in `${APP_PATH}/App/Redux/index.js` now.

File: plugin.js [hunk 3]
```diff
@@ -146,6 +146,27 @@ const remove = async function (context) {
...
+  // unpatch Redux/index.js
+  ignite.patchInFile(`${APP_PATH}/App/Redux/index.js`, {
+    delete: `import { persistReducer } from 'redux-persist'\n`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Redux/index.js`, {
+    delete: `import ReduxPersist from '../Config/ReduxPersist'\n`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Redux/index.js`, {
+    delete: `
+    let finalReducers = reducers
```
Comment (ryanlntn) [inline plugin.js]: You'll want to update whitespace here as well.

File: templates/Rehydration.js [hunk 4]
```diff
@@ -6,7 +6,6 @@ import DebugConfig from '../Config/DebugConfig'
 
 const updateReducers = (store: Object) => {
   const reducerVersion = ReduxPersist.reducerVersion
-  const config = ReduxPersist.storeConfig
```
Comment (ryanlntn) [inline templates/Rehydration.js:9]: Can you explain this change? Do we not need to apply the config here now?

File: plugin.js [hunk 5]
```diff
@@ -146,6 +146,27 @@ const remove = async function (context) {
...
+    // If rehydration is on use persistReducer otherwise default combineReducers
+    if (ReduxPersist.active) {
+      const persistConfig = ReduxPersist.storeConfig
+      finalReducers = persistReducer(persistConfig, reducers)
+    }\n`
+  })
+  ignite.patchInFile(`${APP_PATH}/App/Containers/RootContainer.js`, {
```
Comment (ryanlntn) [inline plugin.js]: And this will need to be updated to `${APP_PATH}/App/Redux/index.js` too.

**Review by ryanlntn — CHANGES_REQUESTED** [review]
@Amurmurmur Thanks for this! There are a couple issues I'm seeing when igniting a new app. Please see my comments.

File: templates/Rehydration.js (hunk 4 above)
Comment (ryanlntn) [reply templates/Rehydration.js:9]: Nevermind. I just checked out the migration guide https://github.com/rt2zz/redux-persist#migration-from-v4-to-v5

**Comment by Amurmurmur** [comment]
@ryanlntn Please check the changes. Thank you very much for pointing them out 👍

**Review by ryanlntn — APPROVED** [review]
Awesome! Thanks @Amurmurmur

**Pull request merged**